### PR TITLE
WIP Unixsocket experiment

### PR DESCRIPTION
**Pull request by KostyaSha**
This is a work in progress branch with experimental use of https://github.com/jnr/jnr-unixsocket and netty. The goal is to implement a `java.nio.channel.SocketChannel` than can be used by netty to support unix domain sockets on linux and macos.

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/692)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/netty/DockerCmdExecFactoryImpl.java [hunk 1]
```diff
@@ -216,22 +214,50 @@ private DuplexChannel connect(final Bootstrap bootstrap) throws InterruptedExcep
     }
 
     private class UnixDomainSocketInitializer implements NettyInitializer {
+
+        final java.io.File path = new java.io.File("/var/run/docker.sock");
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/netty/DockerCmdExecFactoryImpl.java]: import normally

File: src/test/java/com/github/dockerjava/netty/exec/InfoCmdExecTest.java [hunk 2]
```diff
@@ -55,7 +55,9 @@ public void info() throws DockerException {
             dockerClient.startContainerCmd(container.getId()).exec();
         }
 
+        System.out.println(">>>>>>>>>>>>>>>>>>>>>");
         Info dockerInfo = dockerClient.infoCmd().exec();
+        System.out.println("<<<<<<<<<<<<<<<<<<<<<");
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/netty/exec/InfoCmdExecTest.java]: noooo no system out

**Comment by KostyaSha** [comment]
rebase

**Comment by marcuslinke** [comment]
@KostyaSha It makes no sense to review this branch. Its experimental not for merging.

**Comment by KostyaSha** [comment]
Even if it experiment i want see what is changed here. Either delete branch either keep PR open.

**Comment by KostyaSha** [comment]
Or experiment in your fork.

**Comment by marcuslinke** [comment]
Accepted. But review is not needed at this point.

**Comment by KostyaSha** [comment]
> Accepted. But review is not needed at this point.

I would strongly recommend rebase because travis builds was reworked.

**Comment by marcuslinke** [comment]
closed in favor of https://github.com/docker-java/docker-java/pull/694

**Pull request closed without merging**